backend/tournesol/suggestions/graph.py

```diff
@@ -327,18 +326,14 @@ def compute_information_gain(self, scaling_factor_increasing_videos: list[Sugges
                         if eigenvalues[1] > self.LAMBDA_THRESHOLD:
                             u_index = sg.uid_to_index[vb.uid]
                             v_index = sg.uid_to_index[va.uid]
-                            va._graph_sparsity_score = sg.similarity_matrix[u_index, v_index]
+                            va._graph_sparsity_score[vb] = sg.similarity_matrix[u_index, v_index]
                         elif va.uid not in sg.uid_to_index:
-                            va._graph_sparsity_score = 1
+                            va._graph_sparsity_score[vb] = 1
                         else:
-                            va._graph_sparsity_score = 0
+                            va._graph_sparsity_score[vb] = 0
```

Comment:
The first condition `if eigenvalues[1] > self.LAMBDA_THRESHOLD` is never reached in the test suite.

`np.linalg.eigvalsh` returns the eigenvalues in _ascending_ order. Is it deliberate to use the second smallest value?

Reply:
Thanks for noticing, I changed it to get the correct eigenvalue